### DISCUSSION TITLE
Edge runtime: Module not found: Can't resolve 'fs' on build

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/next.js#38861] [Help]
My application has been running serverless for a very long time. I'm aware that when a function is excecuted it cannot use the `fs` node package because there is no Filesystem to access, however this limitation usually doesn't impact using `fs` during build time.

I was trying to move my functions to use the Edge runtime, but my build keeps failing because some packages I'm using import `fs` at some point in their codebase.

```
https://nextjs.org/docs/messages/module-not-found

./node_modules/browser-sync-ui/lib/UI.js:1:0
Module not found: Can't resolve 'fs'

Import trace for requested module:
./node_modules/browser-sync-ui/index.js
./node_modules/browser-sync/dist/browser-sync.js
./node_modules/browser-sync/dist/index.js
```

I want to make clear that **`fs` is never called at runtime**, the build is failing just because `fs` is mentioned in the package's codebase.

Is this how it is intended to work? How come serverless functions allow for `fs` during build but Edge doesn't?

I'm also getting:
```
> webpack config.resolve.alias was incorrectly overridden. https://nextjs.org/docs/messages/invalid-resolve-alias
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

But I haven't overridden my webpack config at all:
```
module.exports = (phase, { defaultConfig }) => {
  /**
   * @type {import('next').NextConfig}
   */
  const nextConfig = {
    reactStrictMode: true,
    images: {
      domains: ["pbs.twimg.com"],
    },
    async rewrites() {
      return [
        {
          source: "/bee.js",
          destination: "https://cdn.splitbee.io/sb.js",
        },
        {
          source: "/_hive/:slug",
          destination: "https://hive.splitbee.io/:slug",
        },
      ];
    },
  };
  return nextConfig;
};
```

**Comment by riqwan** [comment]
I know this is pretty old, but you need to tell webpack to push this dependency out during build.
```
webpack: (
    config,
    { buildId, dev, isServer, defaultLoaders, nextRuntime, webpack },
  ) => {
    config.externals.push({
      "fs": "fs"
    })
}
```
